### PR TITLE
[CP-1007] Add API for rebooting to USB MSC mode

### PR DESCRIPTION
**Pull request by pawel-mudita**
It will allow to reboot the device to USB MSC mode
without need to interact with a physical keyboard.

### CODE REVIEW

**Review by wojtekrzepecki — APPROVED** [review]
LGTM

File: module-services/service-appmgr/model/ApplicationManagerCommon.cpp [hunk 1]
```diff
@@ -157,14 +157,13 @@ namespace app::manager
...
         switch (closeReason) {
         case sys::CloseReason::SystemBrownout:
-            [[fallthrough]];
         case sys::CloseReason::LowBattery:
             act = ActionRequest{this->GetName(), app::manager::actions::SystemBrownout, nullptr};
             break;
         case sys::CloseReason::RegularPowerDown:
-            [[fallthrough]];
         case sys::CloseReason::Reboot:
-            [[fallthrough]];
```
Comment (SP2FET) [inline module-services/service-appmgr/model/ApplicationManagerCommon.cpp:167]: why did you remove the [[fallthrough]] modifiers?

Comment (pawel-mudita) [reply module-services/service-appmgr/model/ApplicationManagerCommon.cpp:167]: Because they seem to be used incorrectly.
The compiler may warn about unintentional fallthrough, if a `case` is ended without `break` or `[[fallthrough]]` and have at least one statement. With an empty `case` the compiler will not warn.

Comment (SP2FET) [reply module-services/service-appmgr/model/ApplicationManagerCommon.cpp:167]: sure, isocpp says the same ;)
https://isocpp.org/files/papers/P0188R0.pdf

**Review by SP2FET — APPROVED** [review]

**Review by niepiekm — APPROVED** [review]

**Review by lucckb — APPROVED** [review]
